src/libraries/System.IO/tests/Stream/Stream.NullTests.cs

```diff
@@ -131,16 +131,57 @@ public static void TestNullTextReader(TextReader input)
 
             if (sr != null)
                 Assert.True(sr.EndOfStream, "EndOfStream property didn't return true");
-            input.ReadLine();
+            Assert.Null(input.ReadLine());
             input.Dispose();
 
             input.ReadLine();
             if (sr != null)
                 Assert.True(sr.EndOfStream, "EndOfStream property didn't return true");
-            input.Read();
-            input.Peek();
-            input.Read(new char[2], 0, 2);
-            input.ReadToEnd();
+            Assert.Equal(-1, input.Read());
+            Assert.Equal(-1, input.Peek());
+
+            var chars = new char[2];
+            Assert.Equal(0, input.Read(chars, 0, chars.Length));
+            Assert.Equal(0, input.Read(chars.AsSpan()));
+            Assert.Equal(0, input.ReadBlock(chars, 0, chars.Length));
+            Assert.Equal(0, input.ReadBlock(chars.AsSpan()));
+            Assert.Equal("", input.ReadToEnd());
+            input.Dispose();
+        }
+
+        [Theory]
+        [MemberData(nameof(NullReaders))]
+        public static async Task TestNullTextReaderAsync(TextReader input)
+        {
+            var chars = new char[2];
+            Assert.Equal(0, await input.ReadAsync(chars, 0, chars.Length));
+            Assert.Equal(0, await input.ReadAsync(chars.AsMemory(), default));
+            Assert.Equal(0, await input.ReadBlockAsync(chars, 0, chars.Length));
+            Assert.Equal(0, await input.ReadBlockAsync(chars.AsMemory(), default));
+            Assert.Null(await input.ReadLineAsync());
+            Assert.Null(await input.ReadLineAsync(default));
+            Assert.Equal("", await input.ReadToEndAsync());
+            Assert.Equal("", await input.ReadToEndAsync(default));
+            input.Dispose();
+        }
+
+        [Theory]
+        [MemberData(nameof(NullReaders))]
+        public static async Task TestCanceledNullTextReaderAsync(TextReader input)
+        {
+            using var tokenSource = new CancellationTokenSource();
```

Comment:
nit: there is no need to use `var`

```suggestion
            using CancellationTokenSource tokenSource = new ();
```